lib/web_ui/lib/src/engine/window.dart

```diff
@@ -72,11 +73,32 @@ base class EngineFlutterView implements ui.FlutterView {
 
   final ViewConfiguration _viewConfiguration = const ViewConfiguration();
 
+  /// Whether this [EngineFlutterView] has been disposed or not.
+  bool isDisposed = false;
+
+  /// Disposes of the [EngineFlutterView] instance and undoes all of its DOM
+  /// tree and any event listeners.
+  @mustCallSuper
+  void dispose() {
+    isDisposed = true;
+    platformDispatcher.unregisterView(this);
+    dimensionsProvider.close();
+    dom.rootElement.remove();
+    // TODO(harryterkelsen): What should we do about this in multi-view?
+    renderer.clearFragmentProgramCache();
+  }
+
   @override
-  void render(ui.Scene scene) => platformDispatcher.render(scene, this);
+  void render(ui.Scene scene) {
+    assert(!isDisposed, 'Trying to render a disposed EngineFlutterView.');
+    platformDispatcher.render(scene, this);
+  }
 
   @override
-  void updateSemantics(ui.SemanticsUpdate update) => platformDispatcher.updateSemantics(update);
+  void updateSemantics(ui.SemanticsUpdate update) {
+    assert(!isDisposed, 'Trying to update semantics on a disposed EngineFlutterView.');
```

Comment:
(Similar comment to the one in the `render` method)

Reply:
(Similar reply) :P